test/connection-timeout.js

```diff
@@ -7,6 +7,7 @@ const describe = require('mocha').describe
 const it = require('mocha').it
 const before = require('mocha').before
 const after = require('mocha').after
+const connectionFailure = new Error('Temporary connection failure');
```

Comment:
should this be scoped outside of specific test suite?

Reply:
This is re-used by multiple tests, therefor I moved it to the top, but I can also inline it as you prefer.

Reply:
There is no single describe. suite above these tests to which it could be scoped?